compiler/lib/eval.ml

```diff
@@ -359,14 +432,24 @@ let eval_instr info ((x, loc) as i) =
                 ( x
                 , Prim
                     ( prim
-                    , List.map2 prim_args prim_args' ~f:(fun arg c ->
-                          match c with
-                          | Some ((Int _ | Float _ | NativeString _) as c) -> Pc c
-                          | Some (String _ as c) when Config.Flag.use_js_string () -> Pc c
-                          | Some _
+                    , List.map2 prim_args prim_args' ~f:(fun arg (c : constant option) ->
+                          match c, target with
+                          | Some ((Int _ | NativeString _) as c), _ -> Pc c
```

Comment:
NativeString with the Wasm backend should probably be excluded as well if I understand right.

Reply:
We have changed the OCaml physical equality `(==)` to use the JavaScript equality `===` on JavaScript values, so this should actually be correct.

Reply:
Reading `Flow.constant_identical`, is seems that NativeString are boxed. Inlining it here could duplicate allocation similar to Int32 and NativeInt. Am i missing something ?

Reply:
That seems right. I restricted the inlining to Javascript.